### PR TITLE
fix(python): fix as_dict with include_key=False for partition_by

### LINKED ISSUE #9862: partition_by fails when as_dict=True and include_key=False
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```
df.partition_by("name", as_dict=True, include_key=False)
```
```
exceptions.ColumnNotFoundError: name
```

### Issue description

 This is my fault, should have put in a unit test for this case.

### Expected behavior

Should not fail.

### Installed versions

<details>

```
--------Version info---------
Polars:              0.18.7
Index type:          UInt32
Platform:            Windows-10-10.0.19045-SP0
Python:              3.11.2 (tags/v3.11.2:878ead1, Feb  7 2023, 16:38:35) [MSC v.1934 64 bit (AMD64)]

----Optional dependencies----
adbc_driver_sqlite:  <not installed>
connectorx:          0.3.2a6
deltalake:           <not installed>
fsspec:              <not installed>
matplotlib:          3.7.1
numpy:               1.24.2
pandas:              2.0.3
pyarrow:             11.0.0
pydantic:            <not installed>
sqlalchemy:          2.0.7
xlsx2csv:            0.8.1
xlsxwriter:          3.0.9
```

</details>

### PR DESCRIPTION
**Pull request by mcrumiller**
Resolves #9862

This requires an extra call to `unique()` on the key columns. The alternative is to remove the `include_key` from the rust side and only implement this for `as_dict=True`, for which someone could make a good case. As mentioned in the issue, without `as_dict=True`, if the key columns are dropped then the user has no way of knowing which partition corresponds to which label.

### CODE REVIEW

**Comment by s-banach** [comment]
Has anybody looked at this?

File: py-polars/polars/dataframe/frame.py [hunk 1]
```diff
@@ -6865,10 +6865,13 @@ def partition_by(
         ]
 
         if as_dict:
+            df = self._from_pydf(self._df)
             if len(by) == 1:
```
Comment (ritchie46) [inline py-polars/polars/dataframe/frame.py:6869]: Could we do the `unique` call only when `include_key` is `False`?

Comment (mcrumiller) [reply py-polars/polars/dataframe/frame.py]: @ritchie you're right, just pushed a non-unique line when `include_key` is `False`.

**Review by ritchie46 — APPROVED** [review]
Alright. Thank you @mcrumiller

**Pull request merged**